### PR TITLE
Refactor/redux

### PR DESCRIPTION
**Pull request by foolhardy21**
## Redux toolkit.

This PR has the redux toolkit integration. All the `context API` and `useReducer` files have been replaced by redux action creators and `createAsyncThunk`.

- Files to review:
   - [ ] src/slices/

- Use this link for preview - https://deploy-preview-11--remarkable-flan-74a807.netlify.app/login

Thanks

### CODE REVIEW

File: src/slices/auth.slice.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        },
+        updateUserLogInStatus: (state) => {
+            state.isUserLoggedIn = true
+        }
+    },
+    extraReducers: {
+        [logInUser.fulfilled]: (state, action) => {
+            window.localStorage.setItem('userToken', action.payload.data.encodedToken)
+            window.localStorage.setItem('username', action.payload.data.foundUser.username)
+            state.isUserLoggedIn = true
```
Comment (Vishnu-Aithal) [inline src/slices/auth.slice.js:50]: I think all local storage managing should be moved to thunk as it as a side effect

**Review by Vishnu-Aithal — APPROVED** [review]
All Slices Looks Good, 
suggestions - localStorage managing can be moved to thunk
sign up slice can be maintained locally in the component itself

File: src/utils/constants.util.js [hunk 2]
```diff
@@ -1,23 +1,3 @@
...
-export const ACTION_REMOVE_LOADING = 'REMOVE_LOADING'
-export const ACTION_INIT_PROFILE_POSTS = 'INIT_PROFILE_POSTS'
-export const ACTION_REMOVE_POST = 'REMOVE_POST'
-export const ACTION_EDIT_POST = 'EDIT_POST'
-export const ACTION_SET_BIO = 'SET_BIO'
-export const ACTION_UPDATE_BIO = 'UPDATE_BIO'
-
 export const API_LOGIN = '/api/auth/login'
 export const API_SIGNUP = '/api/auth/signup'
 export const API_POSTS = '/api/posts'
```
Comment (Maianki) [inline src/utils/constants.util.js:3]: appreciation: use of constants for API end points

**Review by Maianki — APPROVED** [review]
Good job. Readable and clean code you can go ahead with the mrege.

File: src/slices/auth.slice.js (hunk 1 above)
Comment (foolhardy21) [reply src/slices/auth.slice.js:50]: I've made the changes. Thanks.

**Pull request merged**